src/main.py

```diff
@@ -197,8 +198,39 @@ def commit_changes(changes: t.List[str]) -> CommittedChanges:
         base_branch = None
 
     # Moved to detached HEAD
-    check_call(["git", "checkout", "HEAD@{0}"])
-    check_call(["git", "commit", "-am", message])
+    log.info("Switching to detached HEAD")
+    check_call(["git", "-c", "advice.detachedHead=false", "checkout", "HEAD@{0}"])
+
+    retry_commit = False
+    try:
+        check_call(["git", "commit", "-am", message])
+    except subprocess.CalledProcessError:
+        retry_commit = True
+        pass
+
+    if retry_commit:
+        log.warning("Committing failed. Falling back to a sanitised config")
+        git_email = getpass.getuser() + "@" + "localhost"
+        assert git_email is not None
+        env = {
+            "GIT_CONFIG_GLOBAL": "/dev/null",
+            "GIT_CONFIG_NOSYSTEM": "1",
+            "GIT_CONFIG": "''",
+        }
+        subprocess.run(
+            [
+                "git",
+                "-c",
+                f"user.email={git_email}",
+                "commit",
+                "--no-verify",
+                "--no-gpg-sign",
+                "-am",
+                message,
+            ],
+            check=True,
+            env=env,
+        )
```

Comment:
The point of the `check_call()` wrapper is to log the command being executed, which can be very useful in understanding what the program is doing. I know `subprocess.run()` is the more modern replacement and that `check_call()` is not used everywhere. I guess it is OK to not log the command line. It does have a computed parameter…